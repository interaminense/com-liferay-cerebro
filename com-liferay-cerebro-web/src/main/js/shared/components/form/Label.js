import Component, {Config} from 'metal-jsx';

import Icon from '../Icon';

class Label extends Component {
	render() {
		const {children, info, required} = this.props;

		return (
			<label {...this.otherProps()} class="form-control-label label-root">
				{children}

				{info && (
					<Icon
						data-tooltip
						elementClasses="info"
						symbol="question-circle-full"
						title={info}
					/>
				)}

				{required && (
					<span>{`(${Liferay.Language.get('required')})`}</span>
				)}
			</label>
		);
	}
}

Label.PROPS = {
	info: Config.string(),
	required: Config.bool().value(false)
};

export default Label;