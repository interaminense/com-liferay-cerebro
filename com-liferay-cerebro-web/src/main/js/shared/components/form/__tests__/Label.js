import Label from '../Label';

describe('Label', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Label();

		expect(component).toMatchSnapshot();
	});

	it('should render as required', () => {
		component = new Label({required: true});

		expect(component).toMatchSnapshot();
	});
});