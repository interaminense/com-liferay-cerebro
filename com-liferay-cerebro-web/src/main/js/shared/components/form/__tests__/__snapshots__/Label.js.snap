// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render 1`] = `<label class="form-control-label label-root"></label>`;

exports[`Label should render as required 1`] = `
<label class="form-control-label label-root">
  <span>(required)</span>
</label>
`;
